src/Lecture1.hs

```diff
@@ -98,7 +116,11 @@ start position can be considered as zero (e.g. substring from the
 first character) and negative end position should result in an empty
 string.
 -}
-subString start end str = error "TODO"
+subString :: Int->Int->[Char]->[Char]
+subString start end str
+    | end<0 = ""
+    | start<0 = take (end+1) str
+    | otherwise = take ((end + 1) - start) (drop start str)
```

Comment:
Perfect implementation 👍🏻 

Just a stylistic comment, It's recommended to put space around binary operators to make code clearer ✨

Reply:
Will take care of it in the next exercise!